### PR TITLE
Improve loader performance

### PR DESCRIPTION
**Pull request by mjhenkes**
### Summary
So it turns out that the rehype-prisim mdx plugin is non performant and causes 3+ minute webpack time in terra-core. rehype-prisim is used for codeblock syntax highlighting.

This PR removes rehype-prisim build time highlighting and replaces it with prism-react-renderer for client side highlighting.

There are also some small changes to the props table loader and the codeblock loader to avoid reading the file twice if the src file has already been read.

### Additional Details
There were some insignificant differences in how syntax was highlighted but nothing major.
I will pull this branch into terra-core as at test and link the test pr back here. 

edit:
https://github.com/cerner/terra-core/pull/2839

@cerner/terra

[CONTRIBUTORS.md]: https://github.com/cerner/terra-dev-site/blob/master/CONTRIBUTORS.md
[License]: https://github.com/cerner/terra-dev-site/blob/master/License.md

### CODE REVIEW

File: src/mdx/_Code.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+const propTypes = {
+  /**
+   * Children for the component.
+   */
+  children: PropTypes.node,
+  /**
+   * ClassName for the component.
+   */
+  className: PropTypes.string,
+
```
Comment (emilyrohrbough) [inline src/mdx/_Code.jsx]: nit: extra line

File: src/mdx/_Code.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+};
+
+const Code = ({ children, className }) => {
+  const language = (className || '').replace(/language-/, '');
```
Comment (emilyrohrbough) [inline src/mdx/_Code.jsx]: What is this doing?

File: src/mdx/_Code.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  return (
+    // We remove the theme to just use the css theme.
+    <Highlight {...highlightProps} code={children} language={language} theme={undefined}>
```
Comment (emilyrohrbough) [inline src/mdx/_Code.jsx]: Are these default props from `prism-react-renderer` not defined on the Highlight component? Or why are you spreading them on it?

File: src/mdx/_Code.jsx (hunk 2 above)
Comment (mjhenkes) [reply src/mdx/_Code.jsx]: classname is passed in as `language-js` or `language-ts`, this removes the `language-` portion. classname can also be undefined, and if thats the case we can't call replace on an undefined string, so i sub in an empty string.

File: src/mdx/_Code.jsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+   * ClassName for the component.
+   */
+  className: PropTypes.string,
+
+};
+
+const Code = ({ children, className }) => {
+  const language = (className || '').replace(/language-/, '');
+  return (
+    // We remove the theme to just use the css theme.
```
Comment (StephenEsser) [inline src/mdx/_Code.jsx]: Lame

**Pull request merged**